flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/sources/CsvTableSourceFactoryBase.java

```diff
@@ -123,7 +126,15 @@ protected CsvTableSource createTableSource(
 			TableSchema formatSchema = params.getTableSchema(FORMAT_FIELDS);
 			// the CsvTableSource needs some rework first
 			// for now the schema must be equal to the encoding
-			if (!formatSchema.equals(tableSchema)) {
+			// Ignore conversion classes in DataType
+			if (!Arrays
+					.stream(formatSchema.getFieldDataTypes())
+					.map(DataType::getLogicalType)
+					.collect(Collectors.toList())
+					.equals(Arrays
+							.stream(tableSchema.getFieldDataTypes())
+							.map(DataType::getLogicalType)
+							.collect(Collectors.toList()))) {
```

Comment:
Maybe we can have a small utility in the class to get the `LogicalType` List from `TableSchema`? Then we don't need the long method chain.

Reply:
OK